service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -47,6 +49,30 @@ function get_metadata_value() {
     "http://metadata/computeMetadata/v1/$1"
 }
 
+#######################################
+# Set guest attributes on GCE. Used here to log completion status of the script.
+# See https://cloud.google.com/compute/docs/metadata/manage-guest-attributes
+# Arguments:
+#   $1: The guest attribute domain and key IE startup_script/status
+#   $2  The data to write to the guest attribute
+#######################################
+function set_guest_attributes() {
+  curl -s -X PUT --data "$2" \
+    -H "Metadata-Flavor: Google" \
+    "http://metadata.google.internal/computeMetadata/v1/instance/guest-attributes/$1"
+}
```

Comment:
This is optional, but you can clarify the arguments by doing something like

```suggestion
function set_guest_attributes() {
  local attr_key="${1}"
  local attr_value="${2}"

  curl -s -X PUT --data "${attr_value}" \
    -H "Metadata-Flavor: Google" \
    "http://metadata.google.internal/computeMetadata/v1/instance/guest-attributes/${attr_key}"
}
```

Reply:
Good example, Willy. I consider it non-optional. It's an important precedent to set in this file.